x-pack/plugins/security_solution/public/entity_analytics/hooks/use_required_risk_engine_privileges.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { NonEmptyArray } from 'fp-ts/lib/NonEmptyArray';
+import { useMemo } from 'react';
+import { getMissingRiskEnginePrivileges } from '../../../common/entity_analytics/risk_engine';
+import type { RiskEngineIndexPrivilege } from '../../../common/constants';
+import { useRiskEnginePrivileges } from '../api/hooks/use_risk_engine_privileges';
+import type { RiskEngineMissingPrivilegesResponse } from './use_missing_risk_engine_privileges';
+
+export const useRequiredRiskEnginePrivileges = (
```

Comment:
I dont know if this is the best approach.

I went back and forth a bit, but it seemed that we just assume, in the server, that we always need both READ and WRITE for the risk engine. Changing that would involve loads of changes which I'd like to avoid.

This hook is very similar to the `useMissingRiskEnginePrivileges` one, but it allows passing in the list of required privileges, which is more flexible. Again, I didnt want to change all the other places that use the other hook.

Let me know what you think @hop-dev

Reply:
Yeah I think doing it on the client is fine, when I created the API I had thought it was "does the user have all permissions required for the risk engine" hence `has_all_required` (which just mimics the elasticsearch privileges response). 

as for duplicating the hook, why did you decide to have a different hook instead of having `required ` an optional parameter in the existing hook? Wouldnt that limit the changes needed?

Reply:
TLDR: I dont know 😅 
I'm pretty sure I had a reason, but today I can't think of anything other than the name is a bit awkward. I can change that.

Most likely, I started out with a different implementation and didnt wanna change the original one, but then ended up just doing a simple optional parameter.